Mac-App/Backend/Backend/BackendAPI/Middleware/ParseSwiftDepsSideEffects/ParseSwiftDepsSideEffectsSpec.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  ParseSwiftDepsSideEffectsSpec.swift
+//  BackendTests
+//
+//  Created by Daniel GARCÍA on 25.05.20.
+//  Copyright © 2020 Acphut Werkstatt. All rights reserved.
+//
+
+@testable import Backend
+import Nimble
+import Quick
+import ReSwift
+
+final class ParseSwiftDepsSideEffectsSpec: QuickSpec {
+    override func spec() {
+        describe("parseSwiftDepsSideEffects") {
+            let sut = parseSwiftDepsSideEffects
+            var actionRecorder: [Action]!
+            var dispatchFuntion: DispatchFunction!
+
+            var nextActionRecorder: [Action]!
+            var nextActionFunction: DispatchFunction!
+
+            beforeEach {
+                actionRecorder = [Action]()
+                dispatchFuntion = { actionRecorder.append($0) }
+
+                nextActionRecorder = [Action]()
+                nextActionFunction = { nextActionRecorder.append($0) }
+
+                stubbedSwiftDeps = []
+            }
+
+            afterEach {
+                actionRecorder = nil
+                dispatchFuntion = nil
+
+                nextActionRecorder = nil
+                nextActionFunction = nil
+            }
+
+            context("when an unrelated/arbitrary action gets dispatched") {
+                it("does not trigger any action") {
+                    let sutMiddleware = sut(mockYamlParser(from:))
+                    let sideEffect = sutMiddleware(dispatchFuntion) { AppState.initialState }(nextActionFunction)
+                    sideEffect(ArbitraryAction.random)
+
+                    expect(actionRecorder.isEmpty).to(beTrue())
+
+                    expect(nextActionRecorder.isEmpty).to(
+                        beFalse(), description: "Side effects must never swallow actions!"
+                    )
+                }
+            }
+
+            context("when SwiftDepsAction.parseFrom action gets dispatched") {
+                context("given at least one url got parsed") {
+                    it("triggers 1 action") {
+                        let expectedNumberOfActions = 1
+
+                        stubbedSwiftDeps = [SwiftDeps.AppDelegateDeps_Fixture]
+                        let sutMiddleware = sut(mockYamlParser(from:))
+                        let sideEffect = sutMiddleware(dispatchFuntion) { AppState.initialState }(nextActionFunction)
+                        sideEffect(SwiftDepsAction.parseFrom(paths: []))
+
+                        expect(actionRecorder.isEmpty).toEventually(beFalse(), timeout: 0.2)
```

Comment:
Was 0.2 just an arbitrary constant? If so should it be refactored out to a property in the test files so if in the future, if timeout needed to be increased, the change would just be applied to the property itself.

Reply:
That's a good idea, I will do that 👍 

Yeah, I took that number based on the average time it takes for the tests to complete + 1.5 milliseconds of extra time for future iterations (in case the implementation changes & it takes longer, which should not be happen but still it's nice to have a bit of extra time)

Reply:
https://github.com/Thurman1776-/Uncomplect-us/pull/9/commits/4adf7725067a668f624d809d81b0cedd2a67e590